instrumentation/nocode/src/test/java/com/splunk/opentelemetry/instrumentation/nocode/JSPSTest.java

```diff
@@ -0,0 +1,205 @@
+      return param.toString();
+    }
+  }
+
+  public static class TakeIntegerPrimitive {
+    public String take(int param) {
+      return Integer.toString(param);
+    }
+  }
+
+  public static class TakeInteger {
+    public String take(Integer param) {
+      return param.toString();
+    }
+  }
+
+  public static class TakeLongPrimitize {
+    public String take(long param) {
+      return Long.toString(param);
+    }
+  }
+
+  public static class TakeLong {
+    public String take(Long param) {
+      return param.toString();
+    }
+  }
+
+  @Test
+  public void testBooleanLiteralParamTypes() {
+    TakeBooleanPrimitive b = new TakeBooleanPrimitive();
+    TakeBoolean B = new TakeBoolean();
+    TakeObject O = new TakeObject();
+    assertEquals("true", JSPS.evaluate("this.take(true)", b, new Object[0]));
+    assertEquals("false", JSPS.evaluate("this.take(false)", B, new Object[0]));
+    assertEquals("true", JSPS.evaluate("this.take(true)", O, new Object[0]));
+  }
+
+  @Test
+  public void testStringLiteralParamTypes() {
+    TakeString S = new TakeString();
+    TakeObject O = new TakeObject();
+    assertEquals("a", JSPS.evaluate("this.take(\"a\")", S, new Object[0]));
+    assertEquals("a", JSPS.evaluate("this.take(\"a\")", O, new Object[0]));
+  }
+
+  @Test
+  public void testIntegerLiteralParamTypes() {
+    TakeIntegerPrimitive i = new TakeIntegerPrimitive();
+    TakeInteger I = new TakeInteger();
+    TakeLongPrimitize l = new TakeLongPrimitize();
+    TakeLong L = new TakeLong();
+    TakeObject O = new TakeObject();
+    assertEquals("13", JSPS.evaluate("this.take(13)", i, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", I, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", l, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", L, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", O, new Object[0]));
+  }
+
+  @Test
+  public void testWhitespace() {
+    String[] tests =
+        new String[] {
+          "this.get(\"key\").substring(1)",
+          " this.get(\"key\").substring(1)",
+          "this .get(\"key\").substring(1)",
+          "this. get(\"key\").substring(1)",
+          "this.get (\"key\").substring(1)",
+          "this.get( \"key\").substring(1)",
+          "this.get(\"key\" ).substring(1)",
+          "this.get(\"key\") .substring(1)",
+          "this.get(\"key\"). substring(1)",
+          "this.get(\"key\").substring (1)",
+          "this.get(\"key\").substring( 1)",
+          "this.get(\"key\").substring(1 )",
+        };
+    for (String test : tests) {
+      assertEquals("alue", JSPS.evaluate(test, thiz, new Object[] {param0}), test);
+    }
+  }
+
+  public void testManyParams() {
+    Object[] params = new Object[13];
+    Arrays.fill(params, new Object());
+    assertEquals(
+        "java.lang.Object", JSPS.evaluate("param12.getClass().getName()", new Object(), params));
```

Comment:
Makes me instantly wonder what `"param13.getClass().getName()"` would do....do we have coverage for going out of bounds? If not, can we add it?

Reply:
Yeah, that's covered several times over in `testInvalidJspsReturnNull`.